src/main/java/org/jabref/gui/preferences/PreviewTabViewModel.java

```diff
@@ -91,8 +91,9 @@ public PreviewTabViewModel(DialogService dialogService, PreferencesService prefe
         initialPreviewPreferences = preferences.getPreviewPreferences();
 
         sourceTextProperty.addListener((observable, oldValue, newValue) -> {
-            if (getCurrentLayout() instanceof TextBasedPreviewLayout) {
-                ((TextBasedPreviewLayout) getCurrentLayout()).setText(sourceTextProperty.getValue().replace("\n", "__NEWLINE__"));
+            var currentLayout = getCurrentLayout();
+            if (currentLayout instanceof TextBasedPreviewLayout) {
```

Comment:
Does patternmatching with instanceof already work?
```
            if (getCurrentLayout() instanceof TextBasedPreviewLayout currentLayout) {
```

Reply:
Unfortunately in jdk14 it's still a preview feature.